examples/nextjs-with-typescript/components/Video.tsx

```diff
@@ -0,0 +1,22 @@
+'use client';
+import React from 'react';
+// @ts-ignore
+import { useMediaUIDispatch as useDispatch } from './media-chrome-react/hooks/useMediaUIDispatch.js';
+
+const Video = () => {
+  const dispatch = useDispatch();
+  return (
+    <video
+      ref={(videoEl) =>
+        dispatch({ type: 'mediaelementchangerequest', detail: videoEl })
+      }
```

Comment:
> As far as I can tell, this is the only way for the React Provider to know about the media element. As such, would we expect all React developers to wire this up? Or could we simplify with maybe another hook?
> 
> ```js
> import { useMediaRef }
> 
> <video ref={useMediaRef} />
> ```

Yup! We could definitely do that and I think it makes sense since it's a "special" kind of event/action.

> 
> Also, how do we handle multiple players on the same page so that when events are dispatched they get sent to the correct media element?

As long as we don't have nested media ui state context providers, we are good here. We can also easily support more complex cases, but I'd strongly recommend not making that a hard requirement for v1.0.